klient/k8s/resources/resources_test.go

```diff
@@ -237,3 +240,43 @@ func TestListAllPods(t *testing.T) {
 
 	t.Logf("pod list contains %d pods", len(pods.Items))
 }
+
+func TestGetCRDs(t *testing.T) {
+	res, err := New(cfg)
+	if err != nil {
+		t.Fatalf("Error creating new resources object: %v", err)
+	}
+
+	// Register type for the API server.
+	e := gexe.New()
+	p := e.RunProc(`kubectl apply -f ./testdata/projectExample/resourcedefinition.yaml`)
+	if p.Err() != nil {
+		t.Fatalf("Failed to register CRD: %v %v", p.Err(), p.Result())
+	}
+	// Sometimes CRDs need just a bit of time before being ready to use.
+	time.Sleep(5 * time.Second)
+
+	// Create one
+	p = e.RunProc(`kubectl apply -f ./testdata/projectExample/project.yaml`)
+	if p.Err() != nil {
+		t.Fatalf("Failed to create a CRD via yaml: %v %v", p.Err(), p.Result())
+	}
+
+	// See that we can't list it because we don't know the type.
+	ps := &projectExample.ProjectList{}
+	err = res.List(context.TODO(), ps)
+	if err == nil {
+		t.Error("Expected error while listing custom resources before adding it to scheme, but got none")
+	}
+
+	// Register type with klient.
+	if err := projectExample.AddToScheme(res.GetScheme()); err != nil {
```

Comment:
Example of how you'd actually use this. If someone asks, just point them to this line.

Reply:
Ok great, that will work too. Put the onus of scheme registration on tester.